schematic/store/synapse.py

```diff
@@ -100,42 +194,55 @@ def __init__(
 
         self._query_fileview()
 
-    def checkEntityType(self, syn_id): 
+    def _purge_synapse_cache(self, root_dir: str = "/var/www/.synapseCache/"):
         """
-        Check the entity type of a synapse entity
-        return: type of synapse entity 
+        Purge synapse cache if it exceeds 7GB
+        Args:
+            root_dir: directory of the .synapseCache function
+        Returns: 
+            if size of cache reaches 7GB, return the number of files that get deleted
+            otherwise, return the total remaining space (assuming total ephemeral storage is 20GB on AWS )
         """
-        entity = self.syn.get(syn_id)
-        type_entity = str(type(entity))
-
-        if type_entity == "<class 'synapseclient.table.EntityViewSchema'>":
-            return "asset view"
-        elif type_entity == "<class 'synapseclient.entity.Folder'>":
-            return "folder"
-        elif type_entity == "<class 'synapseclient.entity.File'>":
-            return "file"
-        elif type_entity == "<class 'synapseclient.entity.Project'>":
-            return "project"
-        else: 
-            return type_entity
+        # try clearing the cache
+        # scan a directory and check size of files
+        cache = self.syn.cache
+        if os.path.exists(root_dir):
+            # set maximum synapse cache allowed
+            # set maximum ephemeral stroage allowed on AWS
+            maximum_storage_allowed_cache_gb = 7
```

Comment:
I think these storage maximums should be passed as args to the function, with the defaults being set as you have them here. That way in the future its easier for people to update the allowed cache sizes if they want to.

Reply:
Yeah. That's a good idea. I will update the function. Thanks Mialy!